src/OpenTelemetry.Exporter.Geneva/Internal/MsgPack/MsgPackTraceExporter.cs

```diff
@@ -342,6 +351,11 @@ internal int AppendResourceAttributesToBuffer(byte[] buffer, int cursor)
                 }
             }
 
+            if (!isDedicatedField && (this.WithResourceAttributes == null || !this.WithResourceAttributes.Contains(entry.Key)))
```

Comment:
shouldn't we short-circuit this entire method if `this.WithResourceAttributes == null` ?